securebanking-openbanking-uk-rs-simulator-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/backoffice/payment/validation/PaymentConsentValidation.java

```diff
@@ -31,4 +48,153 @@ public abstract class PaymentConsentValidation extends GenericValidations {
      */
     public abstract <T> void validate(T consent);
 
+    /**
+     * Get the error events list to build the error response
+     *
+     * @return list of {@link OBError1}
+     */
+    public List<OBError1> getErrors() {
+        return errors;
+    }
+
+    /**
+     * Clear error event list
+     * @return this
+     */
+    public PaymentConsentValidation clearErrors() {
+        this.errors.clear();
+        return this;
+    }
+
+    protected void validateInstructedAmount(OBWriteDomestic2DataInitiationInstructedAmount instructedAmount) {
+        if (isNull(instructedAmount, "InstructedAmount")) {
+            return;
+        }
+        validateAmount(instructedAmount.getAmount());
+        validateCurrency(instructedAmount.getCurrency());
+    }
+
+
+    protected void validateExchangeRateInformation(OBWriteInternational2DataInitiationExchangeRateInformation exchangeRateInformation, String currencyOfTransfer) {
+        if (Objects.isNull(exchangeRateInformation)) {
+            return;
+        }
+
+        if (isNull(currencyOfTransfer, "CurrencyOfTransfer")) {
+            return;
+        }
+        validateCurrency(currencyOfTransfer);
+
+        if (isNull(exchangeRateInformation.getUnitCurrency(), "UnitCurrency")) {
+            return;
+        }
+        validateCurrency(exchangeRateInformation.getUnitCurrency());
+
+        if (isNull(exchangeRateInformation.getRateType(), "RateType")) {
+            return;
+        }
+        OBExchangeRateType2Code rateType = exchangeRateInformation.getRateType();
+        switch (rateType) {
+            case AGREED -> {
+                if (!exchangeRateInformation.getUnitCurrency().equals(currencyOfTransfer)) {
```

Comment:
Is this validation correct? 
From the spec:
> For an "Agreed" RateType a requested exchange rate is populated in the ExchangeRate field, against the UnitCurrency. I.e, if the UnitCurrency is GBP and CurrencyOfTransfer is USD, then ExchangeRate will be 1.34 (USD to 1 GBP).

https://openbankinguk.github.io/read-write-api-site3/v3.1.10/resources-and-data-models/pisp/international-payment-consents.html#notes

Reply:
Validation changed: `must specify ExchangeRate and ContractIdentification when requesting an Agreed RateType`.